### DISCUSSION TITLE
Recommended fetching method for displaying show live data

### DISCUSSION THREAD

**Discussion by rajpraful** [vercel/next.js#12466] [Help]
Hi all, I am developing a PWA to display live user wise data like list of their services or orders (data that changes every minute).

So, what is the best way to call the API, using getStaticProps or getServersideProps or calling API directly from the client-side?

Before I use to use getInitialProps which calls API on server fro first render and from client-side on revisits. Now how to adopt the same?

**Comment by rafaelalmeidatk** [comment]
Your approach still works, you can use `getStaticProps` or `getServerSideProps` for the initial fetch, then you can keep polling (or you can also use websockets) in the client-side to keep the data updated.

**Comment by timneutkens** [comment]
For vercel.com/dashboard we use https://swr.now.sh for dynamic data.

**Reply by tettoffensive** [reply]
@timneutkens Are you just continually polling? Or is there some way with SWR to use websockets?

**Reply by TommySorensen** [reply]
They are just fetching, no websockets. The SWR library has a lot of cool fetching options. Take at look at the options section https://swr.vercel.app/docs/options
